src/WorkItemMigrator/JiraExport/JiraProvider.cs

```diff
@@ -132,6 +134,19 @@ private async Task<JiraAttachment> DownloadAttachmentAsync(JiraAttachment att, W
             return att;
         }
 
+        private async Task DownloadWithJiraRestClientAsync(string url, string fileFullPath)
+        {
+            var attachmentData = Jira.RestClient.DownloadData(url);
+
+            using (var stream = new MemoryStream(attachmentData))
+            {
+                using (var file = File.Create(fileFullPath))
+                {
+                    await stream.CopyToAsync(file);
+                }
+            }
+        }
+
         private void EnsurePath(string path)
         {
```

Comment:
consider changing this a bit, Directory.CreateDirectory will create a full path (c:\temp\a\b\c) so there is no need to traverse the parent/child directories for creation, 
the Directory.Exists(dir) does not need to be there as the Directory.CreateDirectory will check if some or all parts of the path are missing, and if the full path exists it will not throw an exception.

Reply:
Thanks for the input :)